simple_history/utils.py

```diff
@@ -123,3 +125,19 @@ def bulk_update_with_history(
             default_user=default_user,
             default_change_reason=default_change_reason,
         )
+
+
+def get_change_reason_from_object(obj):
+    if hasattr(obj, "_change_reason"):
+        return getattr(obj, "_change_reason")
```

Comment:
This could be `return obj._change_reason`!
(or if this is called many times in loops, even more efficient would be `try: return obj.change_reason    except AttributeError: pass`)

Reply:
Agree on the latter - though this will be removed in next version anyway. How could this be `return obj._change_reason` though? If that doesn't exist, it needs to check `changeReason`

Reply:
I was saying two things at once!

1:  not reason to use getattr + string after you’ve already tested that the attribute exists
```python
if hasattr(obj, "_change_reason"):
    return obj._change_reason
```

2: if performance matters, you can avoid getting the attribute multiple times (hasattr under the covers tries to get the attr and catches AttributeError):
```python
try:
    return object._change_reason
except AttributeError:
   pass

# rest of the code that tries changeReason
```

Reply:
Ah yes, good points.